src/compile/layer.ts

```diff
@@ -117,19 +117,14 @@ export class LayerModel extends Model {
       if (true) { // if shared/union scale
         keys(child.component.scales).forEach(function(channel) {
           let childScales: ScaleComponents = child.component.scales[channel];
-          if (!childScales) {
-            // the child does not have any scales so we have nothing to merge
+          const modelScales: ScaleComponents = scaleComponent[channel];
+
+          if (!childScales || isSignalRefDomain(childScales.main.domain) || modelScales && isSignalRefDomain(modelScales.main.domain)) {
```

Comment:
Yes, but they won't be in the current logic. This needs more work so I just made it work for now.